src/fields.jl

```diff
@@ -52,3 +52,215 @@ Field(LX, data, grid::Grid1D) = Field1D(LX, data, grid)
 Constructs a 2D field of `data` at location `(LX, LY)` on `grid`.
 """
 Field(LX, LY, data, grid::Grid2D) = Field2D(LX, LY, data, grid)
+
+############################################################################
+#-------------------------------Interpolation------------------------------#
+
+𝐼xᶠ(i::Int, f::Field1D{Face})   = (f.data[i] + f.data[i+1]) / 2
+𝐼xᶜ(i::Int, f::Field1D{Centre}) = (f.data[i] + f.data[i-1]) / 2
+
+𝐼xᶠ(i::Int, j::Int, f::Field2D{Face, Centre})   = (f.data[i, j] + f.data[i+1, j]) / 2
+𝐼xᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i-1, j]) / 2
+𝐼yᶠ(i::Int, j::Int, f::Field2D{Centre, Face})   = (f.data[i, j+1] + f.data[i, j]) / 2
+𝐼yᶜ(i::Int, j::Int, f::Field2D{Centre, Centre}) = (f.data[i, j] + f.data[i, j-1]) / 2
+
+"""
+    𝐼x!(output::Field1D{Centre}, input::Field1D{Face})
+
+Interpolates a 1D field of `data` from Face to Centre.
```

Comment:
```suggestion
Interpolates a 1D `input` field that lives in `Face`s to `output` field that lives on `Centre`s.
```

I just removed `data` from the docstring since `data` is not part of anything here. Instead I mentioned `input` and `output` which are the actual function's arguments.

Reply:
Let's make the docstrings more useful. Think of the pure PhD students of 2034 who will be struggling to reproduce the results by Bhagtani et al 2022... :)

Reply:
Sure. Just a remark: this PR isn't completed yet. My main aim through the previous commit was to ask you about the broken code.

Reply:
Sure. But you said one test should pass. It was not passing for me.

Reply:
Hm.... This was committed but then you changed it back? Do you disagree with my suggestion?